src/sentry/api/helpers/group_index/update.py

```diff
@@ -780,12 +781,21 @@ def handle_has_seen(
         GroupSeen.objects.filter(group__in=group_ids, user_id=user_id).delete()
 
 
-def handle_priority(priority: str, group_list: Sequence[Group], actor: User) -> None:
+def handle_priority(
+    priority: str, group_list: Sequence[Group], actor: User, project_lookup: dict[int, Project]
+) -> None:
     for group in group_list:
+        priority_value = PriorityLevel.from_str(priority) if priority else None
+
+        if priority_value is None or group.priority == priority_value:
```

Comment:
Also ensure that we only activate this logic when the priority actually changes

Reply:
Does it make sense to push that check to `update_priority` so it's not the caller's responsibility each time?